File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/api/dto/RequestTypeResponse.java
package br.ufpr.sept.so2.modules.solicitacoes.api.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;
import java.util.UUID;

public record RequestTypeResponse(
        UUID id,
        String codigo,
        String nome,
        String descricao,
        String status,
        int prazoDias,
        int versao,
        Map<String, Object> formSchema,
        Map<String, Object> workflowJson,
        @JsonProperty("_links") Map<String, String> links
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/api/dto/RequestTypeResponse.kt
package br.ufpr.sept.so2.modules.solicitacoes.api.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class RequestTypeResponse(
    val id: UUID,
    val codigo: String,
    val nome: String,
    val descricao: String?,
    val status: String,
    val prazoDias: Int,
    val versao: Int,
    val formSchema: Map<String, Any?>,
    val workflowJson: Map<String, Any?>,
    @get:JsonProperty("_links")
    val links: Map<String, String>,
)
